tests/evaluate/test_default_evaluator.py

```diff
@@ -2927,3 +2928,91 @@ def test_evaluate_with_latency():
         "token_count",
     }
     assert all(isinstance(grade, float) for grade in logged_data["latency"])
+
+
+properly_formatted_openai_response1 = {
+    "candidates": [
+        {
+            "text": '{\n  "score": 3,\n  "justification": "' "justification" '"\n}',
+            "metadata": {"finish_reason": "stop"},
+        }
+    ],
+    "metadata": {
+        "input_tokens": 569,
+        "output_tokens": 93,
+        "total_tokens": 662,
+        "model": "gpt-3.5-turbo-0613",
+        "route_type": "llm/v1/completions",
+    },
+}
+
+
+def test_evaluate_with_correctness():
```

Comment:
I tried this code on `master` just to confirm that it reproduces the issue, and it does. While repro-ing the issue, I saw this message: 

```
Below are the existing column names for the input/output data:
Input Columns: ['inputs']
Output Columns: []
Note that this does not include the output column: 'output'
```

The part about
```
Note that this does not include the output column: 'output'
```

is kind of confusing. Why not include `output` in the `Output columns` in the error message? It is an output column :)

cc @annzhang-db @sunishsheth2009

Reply:
@annzhang-db @sunishsheth2009 it looks like the name of the output column in the error message is incorrect too. For example, I ran the following code with `ground_truthz` (intentional typo) as the grading context column. The error message mentioned the output column `output`, but the output column is actually called `predictions`.

```
import mlflow.metrics
import mlflow
import pandas as pd

import os
os.environ["OPENAI_API_KEY"] = "<REDACTED>"

metric = mlflow.metrics.make_genai_metric(
      name="correctness",
      definition=(
          "Correctness refers to how well the generated output matches "
          "or aligns with the reference or ground truth text that is considered "
          "accurate and appropriate for the given input. The ground truth serves as "
          "a benchmark against which the provided output is compared to determine the "
          "level of accuracy and fidelity."
      ),
      grading_prompt=(
          "Correctness: If the answer correctly answer the question, below "
          "are the details for different scores: "
          "- Score 0: the answer is completely incorrect, doesn’t mention anything about "
          "the question or is completely contrary to the correct answer. "
          "- Score 1: the answer provides some relevance to the question and answer "
          "one aspect of the question correctly. "
          "- Score 2: the answer mostly answer the question but is missing or hallucinating "
          "on one critical aspect. "
          "- Score 4: the answer correctly answer the question and not missing any "
          "major aspect"
      ),
      examples=[],
      version="v1",
      model="openai:/gpt-4",
      grading_context_columns=["ground_truthz"],
      parameters={"temperature": 0.0},
      aggregations=["mean", "variance", "p90"],
      greater_is_better=True,
  )

with mlflow.start_run():
    eval_df = pd.DataFrame(
        {
            "inputs": [
                "What is MLflow?",
                "What is Spark?",
                "What is Python?",
            ],
            "ground_truth": [
                "MLflow is an open-source platform",
                "Apache Spark is an open-source, distributed computing system",
                "Python is a high-level programming language",
            ],
            "prediction": [
                "MLflow is an open-source platform",
                "Apache Spark is an open-source, distributed computing system",
                "Python is a high-level programming language",
            ],
        }
    )
    results = mlflow.evaluate(
        data=eval_df,
        evaluators="default",
        targets="ground_truth",
        predictions="prediction",
        extra_metrics=[metric],
    )

    print(results.metrics)
    print(results.tables["eval_results_table"].to_dict().keys())
```

Reply:
@annzhang-db @sunishsheth2009 I also found that `col_mapping` is broken with `targets`. The following should work, but it doesn't:

```
import mlflow.metrics
import mlflow
import pandas as pd

import os
os.environ["OPENAI_API_KEY"] = "<REDACTED>"

metric = mlflow.metrics.make_genai_metric(
      name="correctness",
      definition=(
          "Correctness refers to how well the generated output matches "
          "or aligns with the reference or ground truth text that is considered "
          "accurate and appropriate for the given input. The ground truth serves as "
          "a benchmark against which the provided output is compared to determine the "
          "level of accuracy and fidelity."
      ),
      grading_prompt=(
          "Correctness: If the answer correctly answer the question, below "
          "are the details for different scores: "
          "- Score 0: the answer is completely incorrect, doesn’t mention anything about "
          "the question or is completely contrary to the correct answer. "
          "- Score 1: the answer provides some relevance to the question and answer "
          "one aspect of the question correctly. "
          "- Score 2: the answer mostly answer the question but is missing or hallucinating "
          "on one critical aspect. "
          "- Score 4: the answer correctly answer the question and not missing any "
          "major aspect"
      ),
      examples=[],
      version="v1",
      model="openai:/gpt-4",
      grading_context_columns=["ground_truthz"],
      parameters={"temperature": 0.0},
      aggregations=["mean", "variance", "p90"],
      greater_is_better=True,
  )

with mlflow.start_run():
    eval_df = pd.DataFrame(
        {
            "inputs": [
                "What is MLflow?",
                "What is Spark?",
                "What is Python?",
            ],
            "ground_truth": [
                "MLflow is an open-source platform",
                "Apache Spark is an open-source, distributed computing system",
                "Python is a high-level programming language",
            ],
            "prediction": [
                "MLflow is an open-source platform",
                "Apache Spark is an open-source, distributed computing system",
                "Python is a high-level programming language",
            ],
        }
    )
    results = mlflow.evaluate(
        data=eval_df,
        evaluators="default",
        targets="ground_truthz",
        predictions="prediction",
        extra_metrics=[metric],
        evaluator_config={
            "col_mapping": {
                "ground_truthz": "ground_truth"
            }
        }
    )

    print(results.metrics)
    print(results.tables["eval_results_table"].to_dict().keys())
```

Reply:
Case 1 and 2 should be resolved in this PR: https://github.com/mlflow/mlflow/pull/9983#pullrequestreview-1686585155
And adding solution for case 3 in this PR #9705